semantics.json

```diff
@@ -20,6 +20,51 @@
     }
   },
   {
+    "name": "icon",
+    "type": "select",
+    "label": "Hotspot Icon",
+    "importance": "medium",
+    "description": "The icon of the hotspots",
+    "options": [
+      {
+        "value": "plus",
+        "label": "Plus"
+      },
+      {
+        "value": "minus",
+        "label": "minus"
+      },
+      {
+        "value": "times",
+        "label": "Times"
+      },
+      {
+        "value": "check",
+        "label": "Check"
+      },
+      {
+        "value": "question",
+        "label": "Question"
+      },
+      {
+        "value": "info",
+        "label": "Info"
+      },
+      {
+        "value": "exclamation",
+        "label": "Exclamation"
+      }
+    ],
+    "default": "plus"
+  },
+  {
+    "name": "iconImage",
+    "type": "image",
+    "label": "Icon image override",
+    "optional": true,
+    "description": "Override the existing icon for hotspot and use your own. This overrides the Hotspot Color & Hotspot Icon settings.<br>75px by 75px is recommended for your image."
+  },
+  {
```

Comment:
It's not easy for you to know, but it is possible to show and hide fields based on rules. I have taken the time to make this example for you:
```
{
    "name": "iconType",
    "type": "select",
    "label": "Hotspot Icon",
    "options": [
      {
        "value": "icon",
        "label": "Predefined icon"
      },
      {
        "value": "image",
        "label": "Uploaded image"
      }
    ],
    "default": "icon"
  },
  {
    "name": "icon",
    "type": "select",
    "label": "Hotspot Icon",
    "importance": "medium",
    "description": "The icon of the hotspots",
    "widget": "showWhen",
    "showWhen": {
      "rules": [
        {
          "field": "iconType",
          "equals": [
            "icon"
          ]
        }
      ]
    },
    "options": [
      {
        "value": "plus",
        "label": "Plus"
      },
      {
        "value": "minus",
        "label": "minus"
      },
      {
        "value": "times",
        "label": "Times"
      },
      {
        "value": "check",
        "label": "Check"
      },
      {
        "value": "question",
        "label": "Question"
      },
      {
        "value": "info",
        "label": "Info"
      },
      {
        "value": "exclamation",
        "label": "Exclamation"
      }
    ],
    "default": "plus"
  },
  {
    "name": "iconImage",
    "type": "image",
    "label": "Icon image override",
    "optional": true,
    "description": "Override the existing icon for hotspot and use your own. This overrides the Hotspot Color & Hotspot Icon settings.<br>75px by 75px is recommended for your image.",
    "widget": "showWhen",
    "showWhen": {
      "rules": [
        {
          "field": "iconType",
          "equals": [
            "image"
          ]
        }
      ]
    }
  },
```
Here I introduce a list where the user may chose between "Predefined icon" (default)  or "Uploaded image". The two fields you have added will be shown only when it is relevant.

To be able to use the showWhen widget, you need to add the following to editorDependencies (in library.json):
```
{
  "machineName": "H5PEditor.ShowWhen",
  "majorVersion": 1,
  "minorVersion": 0
}
```